### ISSUE TITLE
Add warning in migration process if there are existing entities on the site

### ISSUE THREAD

**Issue by steve-taylor-wsu** [MukurtuCMS/Mukurtu-CMS#135]
Because we are preserving entity IDs during migration, we should warn the user if they've created any entities they will likely be overwritten during migration.

Entity types to check:
* node
* media
* community
* protocol
* file

Anything else? I don't think we need to check the more "dependent" types like paragraphs.

**Comment by michael-wynne-wsu** [comment]
Nodes encompass everything I can think off not already listed...

**Comment by steve-taylor-wsu** [comment]
@nick-deer Take a stab at this? If any entities of a listed type exist (let's forget about files, just use the other 4), we want to display a message warning the user their content will be overwritten. This should be on the form where the user executes the migration. A general warning is fine, we don't need to call out individual items.

**Comment by nick-deer** [comment]
@michael-wynne-wsu First pass on this is done!

**Comment by michael-wynne-wsu** [comment]
@nick-deer I don't have anything queued up to run a migration test right now. I'll check with Brian, but any chance you could share screenshots or something in the meantime?

**Comment by michael-wynne-wsu** [comment]
Actually - this warning?
![Screenshot 2023-12-05 at 11 57 33 AM](https://github.com/MukurtuCMS/Mukurtu-CMS/assets/88502274/9e8b894c-87fd-4f18-8631-ad23bcbe1dfc)

**Comment by michael-wynne-wsu** [comment]
Interestingly, it did NOT overwrite the community, protocol, DH item, and image I created to test.

**Comment by nick-deer** [comment]
Yeah it was that warning! And huh interesting... I'll have to look into why it didn't overwrite those. We probably don't want to display a message like this if it doesn't actually end up overwriting anything, what do you think?

**Comment by michael-wynne-wsu** [comment]
Let's discuss wednesday, I can show you what I saw.

**Comment by michael-wynne-wsu** [comment]
Decision made to disallow users to migrate into a site with ANY content.

**Comment by steve-taylor-wsu** [comment]
@nick-deer My first thoughts on how I'd do this:
* Move your entityQuery code from ReviewForm to the base form or OverviewForm.
* Have it run in the constructor (also you could likely terminate earlier, no need to run all the queries if any of them return results. Also consider setting the limit to 1 or a low number, no need to return all the site IDs) and populate a boolean and/or create a method for it (e.g., siteHasContent).
* Don't show the submit button on OverviewForm (this is created in the base form so this would be a conditional or an unset depending on where you decide to implement) if site has content
* All forms other than OverviewForm will need a different access function that also checks for existing content because we won't want people to visit any of the other forms directly in that case

This only covers the forms. We haven't tried it yet, but it'd be worth discussing if we want to prevent our migrations from running via Drush in this same circumstance or if command line users are trusted to know what they are doing.

**Comment by michael-wynne-wsu** [comment]
@nick-deer excellent! 

Request: can it be styled to be more prominent (I'm thinking the generic red error message)?

Question: it would be good to test and make sure that deleted content doesn't impact anything in migration just in case. I should be able to test that on my end.

**Comment by nick-deer** [comment]
@michael-wynne-wsu I styled the error message!
![image](https://github.com/MukurtuCMS/Mukurtu-CMS/assets/43937352/6e152a64-6369-475a-902f-1522ec380d78)
I also tested migrating after deleting site content and the migration seemed to work just fine! Let me know how the migration tests go on your end

**Comment by michael-wynne-wsu** [comment]
That is perfect, thank you!

I trust your tests, but I'll run it again on Friday just so I can see it all live myself.

**Comment by michael-wynne-wsu** [comment]
Message and denying access if there is content present worked as desired. And deleting that content then opened up the migrator which ran as expected.

Thanks!
